agent/eventhandler/handler_test.go

```diff
@@ -16,185 +16,84 @@ package eventhandler
-	return "", nil
-}
-func (m *MockECSClient) DiscoverPollEndpoint(string) (string, error) {
-	return "", nil
-}
-func (m *MockECSClient) SubmitTaskStateChange(change api.TaskStateChange) error {
-	return m.submitTaskStateChange(change)
-}
-func (m *MockECSClient) SubmitContainerStateChange(change api.ContainerStateChange) error {
-	return m.submitContainerStateChange(change)
-}
-func (m *MockECSClient) DiscoverTelemetryEndpoint(string) (string, error) {
-	return "", nil
-}
-
-func mockClient(task taskChangeFn, cont containerChangeFn) api.ECSClient {
-	return &MockECSClient{
-		task, cont,
-	}
-}
-
 func contEvent(arn string) api.ContainerStateChange {
 	return api.ContainerStateChange{TaskArn: arn, ContainerName: "containerName", Status: api.ContainerRunning}
 }
 func taskEvent(arn string) api.TaskStateChange {
 	return api.TaskStateChange{TaskArn: arn, Status: api.TaskRunning}
 }
 
-func TestSendsEvents(t *testing.T) {
-
-	// These channels will submit "successful" state changes back to the test
-	taskStatus := make(chan api.TaskStateChange)
-	contStatus := make(chan api.ContainerStateChange)
-
-	// These counters let us know how many errors have happened of each type
-	var taskRetriableErrors, contRetriableErrors, taskUnretriableErrors, contUnretriableErrors, taskCalls, contCalls int32
-	resetCounters := func() {
-		taskCalls = 0
-		contCalls = 0
-		taskRetriableErrors = 0
-		contRetriableErrors = 0
-		taskUnretriableErrors = 0
-		contUnretriableErrors = 0
-	}
-	resetCounters()
-
-	// These channels are used to tell the mock functions if they should error
-	// or not
-	taskError := make(chan utils.RetriableError)
-	contError := make(chan utils.RetriableError)
-
-	// premade errors for ease of testing
-	retriable := utils.NewRetriableError(utils.NewRetriable(true), errors.New("test"))
-
-	client := mockClient(
-		func(change api.TaskStateChange) error {
-			atomic.AddInt32(&taskCalls, 1)
-			err := <-taskError
-			if err == nil {
-				taskStatus <- change
-				return err
-			}
-			if err.Retry() {
-				atomic.AddInt32(&taskRetriableErrors, 1)
-				return err
-			}
-
-			atomic.AddInt32(&taskUnretriableErrors, 1)
-			return err
-		},
-		func(change api.ContainerStateChange) error {
-			atomic.AddInt32(&contCalls, 1)
-			err := <-contError
-			if err == nil {
-				contStatus <- change
-				return err
-			}
-			if err.Retry() {
-				atomic.AddInt32(&contRetriableErrors, 1)
-				return err
-			}
-			atomic.AddInt32(&contUnretriableErrors, 1)
-			return err
-		},
-	)
+func TestSendsEventsOneContainer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_api.NewMockECSClient(ctrl)
 
 	// Trivial: one container, no errors
+	contCalled := make(chan struct{})
+	taskCalled := make(chan struct{})
+	contEvent1 := contEvent("1")
+	contEvent2 := contEvent("2")
+	taskEvent2 := taskEvent("2")
+
+	client.EXPECT().SubmitContainerStateChange(contEvent1).Do(func(interface{}) { contCalled <- struct{}{} })
```

Comment:
The order of the two `SubmitContainerStateChange` calls is non-deterministic, so `gomock.InOrder` fails due to that.